cluster/examples/kubernetes/ceph/cluster-with-drive-groups.yaml

```diff
@@ -0,0 +1,106 @@
+#################################################################################################################
+# Define the settings for the rook-ceph cluster with common settings for a production cluster.
+# All nodes with available raw devices will be used for the Ceph cluster. At least three nodes are required
+# in this example. See the documentation for more details on storage settings available.
+
+# For example, to create the cluster:
+#   kubectl create -f common.yaml
+#   kubectl create -f operator.yaml
+#   kubectl create -f cluster-with-drive-groups.yaml
+#################################################################################################################
+
+apiVersion: ceph.rook.io/v1
+kind: CephCluster
+metadata:
+  name: rook-ceph
+  namespace: rook-ceph
+spec:
+  cephVersion:
+      image: ceph/ceph:v15.2.3
+      allowUnsupported: false
+  skipUpgradeChecks: false
+  dataDirHostPath: /var/lib/rook
+  mon:
+    count: 3
+    allowMultiplePerNode: false
+  mgr:
+    modules:
+    - name: pg_autoscaler
+      enabled: true
+  dashboard:
+    enabled: true
+    ssl: true
+  crashCollector:
+    disable: false
+  annotations:
+  resources:
+  removeOSDsIfOutAndSafeToRemove: false
+  storage:
+    # When managing a Rook/Ceph cluster's OSD layouts with Drive Groups, the `storage` config is
+    # mostly ignored. `storageClassDeviceSets` can still be used to create OSDs on PVC, but Rook
+    # will no longer use `storage` configs for creating OSDs on a node's devices. To avoid
+    # confusion, we recommend using the `storage` config OR `driveGroups` and never both.
+    # storageClassDeviceSets: # OSDs on PVCs can still be used with Drive Groups
+
+  # Ceph supports adding devices as OSDs by Ceph Drive Group definitions in later versions of Ceph Octopus (v15.2).
+  # See Ceph Drive Group docs for more info: https://docs.ceph.com/docs/master/cephadm/drivegroups/
+  # Drive Groups cannot be used to configure provisioning of OSDs on PVCs.
+  # A Drive Group is defined by a name, a Ceph Drive Group spec, and a Rook placement
+  driveGroups:
+    # Default example: use all nodes and all devices on those nodes
+    # This is equivalent to `storage:useAllNodes: true` and `storage:useAllDevices: true`
```

Comment:
Are the examples below not sufficient? I thought the basic example set enabled by default would be a good starting point for people trying this out.

Reply:
Yeah, I saw those examples after I made this comment and they are all good examples. But since drive groups are really about flexibility and customization, makes sense to keep this default since anything else is a customization that doesn't make sense as a default.